www/src/views/XMultiCredentialView.tsx

```diff
@@ -0,0 +1,151 @@
+import { useMutation, useQuery } from "@apollo/react-hooks";
+import gql from "graphql-tag";
+import * as React from "react";
+import { FunctionComponent } from "react";
+import { Button, Table } from "semantic-ui-react";
+import { XNoCredentialsFound } from "../components/credential";
+import { XBoundary } from "../components/layout";
+import { XErrorMessage, XLoadingMessage } from "../components/messages";
+import { Credential } from "../graphql/models";
+
+export const MULTI_CREDENTIAL_QUERY = gql`
+  {
+    credentials {
+      id
+      principal
+      secret
+      kind
+      fails
+      target {
+        id
+        name
+      }
+    }
+  }
+`;
+
+const REMOVE_CREDENTIAL = gql`
+  mutation RemoveCredential($credential: ID!) {
+    deleteCredential(input: { id: $credential })
+  }
+`;
+
+type RemoveCredentialButtonProps = {
+  credential: Credential;
+};
+
+const RemoveCredentialButton: FunctionComponent<RemoveCredentialButtonProps> = ({
+  credential
+}) => {
+  const [removeCredential] = useMutation(REMOVE_CREDENTIAL, {
+    refetchQueries: [{ query: MULTI_CREDENTIAL_QUERY }]
+  });
+
+  const handleRemove = () => {
+    let vars = {
+      credential: credential.id
+    };
+
+    removeCredential({
+      variables: vars
+    });
+  };
+
+  return (
+    <Button.Group labeled>
```

Comment:
If we move to the aggregated view as mentioned below, you'll want each target to be a label. On Dismiss, you'll delete the credential based on the target id from the label that was clicked, which enables the user to delete a credential from specific targets.